frontend/src/views/AddProgramAttributesView.svelte

```diff
@@ -0,0 +1,218 @@
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
+    justify-content: center;
+    align-items: stretch;
+    align-content: center;
+  }
+
+  .inputs {
+    flex-grow: 1;
+  }
+
+  .inputs *:first-child {
+    margin-top: 0;
+  }
+
+  .actions {
+    margin-top: 2em;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: nowrap;
+    justify-content: space-evenly;
+    align-items: center;
+    align-content: center;
+  }
+
+  .error {
+    margin-top: 2em;
+  }
+
+  select,
+  option {
+    background-color: var(--main-bg-color);
+    color: inherit;
+    border: 1px solid;
+    border-color: inherit;
+    border-radius: 0;
+    padding-top: 0.5em;
+    padding-bottom: 0.5em;
+    padding-left: 1em;
+    padding-right: 1em;
+    margin-left: 0.5em;
+    margin-right: 0.5em;
+    font-size: inherit;
+    font-family: var(--font);
+    box-shadow: none;
+  }
+
+  select {
+    flex-grow: 1;
+    margin: 0 2ch;
+  }
+
+  option {
+    font-family: monospace;
+  }
+
+  .row {
+    justify-content: space-evenly;
+    align-items: baseline;
+    align-content: center;
+    white-space: nowrap;
+    margin-bottom: 1em;
+  }
+</style>
+
+<script>
+  import {
+    selected,
+    selectedResource,
+    settings,
+    resourceNodeDataMap,
+  } from "../stores.js";
+  import Button from "../utils/Button.svelte";
+  import { onMount } from "svelte";
+  import LoadingText from "../utils/LoadingText.svelte";
+  import { cleanOfrakType } from "../helpers";
+
+  export let modifierView;
+  let errorMessage,
+    ofrakProgramAttributesPromise = new Promise(() => {});
+  let selected_attribute = {};
+
+  async function refreshResource() {
+    // Force tree view children refresh
+    $resourceNodeDataMap[$selected].collapsed = false;
+    $resourceNodeDataMap[$selected].childrenPromise =
+      $selectedResource.get_children();
+
+    // Force resource refresh by getting latest model from backend
+    await $selectedResource.get_latest_model();
+
+    // Force hex view refresh with colors
+    const originalSelected = $selected;
+    $selected = undefined;
+    $selected = originalSelected;
+  }
+
+  async function addProgramAttributes() {
+    if (
+      selected_attribute["isa"] &&
+      selected_attribute["bit_width"] &&
+      selected_attribute["endianness"]
+    ) {
+      modifierView = undefined;
+      let program_attributes = JSON.stringify([
+        "ofrak.core.architecture.ProgramAttributes",
+        {
+          isa: selected_attribute["isa"],
+          sub_isa: selected_attribute["sub_isa"],
+          bit_width: selected_attribute["bit_width"],
+          endianness: selected_attribute["endianness"],
+          processor: selected_attribute["processor"],
+        },
+      ]);
+      await $selectedResource.add_program_attributes(program_attributes);
+      await refreshResource();
+    }
+  }
+
+  onMount(async () => {
+    try {
+      await fetch(`${$settings.backendUrl}/get_all_program_attributes`).then(
```

Comment:
Since this whole function is async, you can await `fetch` here without using a `.then` callback.